google-datacatalog-sisense-connector/src/google/datacatalog_connectors/sisense/prepare/datacatalog_tag_factory.py

```diff
@@ -262,7 +264,19 @@ def __make_tag_for_jaql(self, tag_template: TagTemplate,
         self._set_string_field(tag, 'column',
                                jaql_metadata.get('column') or dim_column)
 
-        self._set_string_field(tag, 'formula', jaql_metadata.get('formula'))
+        formula = jaql_metadata.get(constants.JAQL_FORMULA_FIELD_NAME)
+        context = jaql_metadata.get(constants.JAQL_CONTEXT_FIELD_NAME)
+        if formula and context:
+            human_readable_formula = formula
+            parts = re.findall(r'\[(.*?)]', formula)
+            for part in parts:
+                context_title = context.get(f'[{part}]').get('title')
```

Comment:
formula and context comes from different fields, right?

Isn't it possible for `context.get(f'[{part}]')`, to not find the formula part inside the context? What about do some defensive programming here and prevent it from breaking in case it doesn't return?

Reply:
Yes, the formula and its context come in different fields. I've never seen a missing part in the `context` object, but keeping the code safe is always good, so I've just added a test to make sure the code will not break if it happens. Thanks! 😃